packages/kilt-did/README.md

```diff
@@ -3,6 +3,339 @@
+const serviceEndpoints: EndpointData = {
+  contentHash: 'external-endpoints-content-hash00',
+  contentType: 'application/json',
+  urls: [
+    'https://myendpoint.io?endpoint_id=my_endpoint',
+    'ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu',
+    'ftp://myendpoint.io/endpoints/example.json',
+  ],
+}
+
+// Generate the DID-signed creation extrinsic with the provided keys and service endpoints.
+const { extrinsic, did } = await DidUtils.writeDidFromPublicKeys(
+  keystore,
+  {
+    [KeyRelationship.authentication]: {
+      publicKey: authenticationKeyPublicDetails.publicKey,
+      type: DemoKeystore.getKeypairTypeForAlg(
+        authenticationKeyPublicDetails.alg
+      ),
+    },
+    [KeyRelationship.keyAgreement]: {
+      publicKey: encryptionKeyPublicDetails.publicKey,
+      type: DemoKeystore.getKeypairTypeForAlg(
+        encryptionKeyPublicDetails.alg
+      ),
+    },
+  },
+  serviceEndpoints
+)
+// Will print `did:kilt:4sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar`.
+console.log(did)
+
+await BlockchainUtils.signAndSubmitTx(extrinsic, aliceKiltAccount, {
+  resolveOn,
+})
+
+const fullDid = await DefaultResolver.resolveDoc(did)
+```
+
+## Updating a full DID
+
+Once anchored on the KILT blockchain, a KILT full DID can be updated by signing the operation with a valid authentication key. For instance, the following snippet shows how to update the authentication key of a full DID and set it to a new sr25519 key.
+
+```typescript
+// Generate seed for the new authentication key.
+const newAuthenticationKeySeed = '0xabcdeffedcba'
+
+// Ask the keystore to generate a new keypar to use for authentication.
+const newAuthenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: newAuthenticationKeySeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Create a DID operation to replace the authentication key with the new one generated.
+const didUpdateExtrinsic = await DidChain.getSetKeyExtrinsic(
+  KeyRelationship.authentication,
+  {
+    publicKey: newAuthenticationKeyPublicDetails.publicKey,
+    type: DemoKeystore.getKeypairTypeForAlg(
+      newAuthenticationKeyPublicDetails.alg
+    ),
+  }
+)
+
+// Sign the DID operation using the old DID authentication key.
+// This results in an unsigned extrinsic that can be then signed and submitted to the KILT blockchain.
+const didSignedUpdateExtrinsic = await fullDID.authorizeExtrinsic(
+  didUpdateExtrinsic,
+  keystore as KeystoreSigner<string>
+)
+
+// Submit the DID update tx to the KILT blockchain after signing it with the KILT account specified.
+await BlockchainUtils.signAndSubmitTx(
+  didSignedUpdateExtrinsic,
+  aliceKiltAccount,
+  {
+    resolveOn,
+  }
+)
+```
+
+## Deleting a full DID
+
+Once not needed anymore, it is recommended to remove the DID details from the KILT blockchain. The following snippet shows how to do it:
+
+```typescript
+// Create a DID deletion operation.
+const didDeletionExtrinsic = await DidChain.getDeleteDidExtrinsic()
```

Comment:
C: these function names are hard to read. Might it be better to just have `deleteDidExtrinsic()`?

Reply:
Same with the `getSetKeyExtrinsic()`

Reply:
Maybe even `DidChain.extrinsics.delete(), DidChain.extrinsics.setKey()`?